### PR TITLE
server-lib bugfix: need scopingRoot

### PR DESCRIPTION
**Pull request by samouri**
Followup fix to a lil bug in https://github.com/ampproject/worker-dom/pull/1096.
Note: the bug only affects server-lib.

---

Followup: need to come up with a good system for parameterizing tests

### CODE REVIEW

File: src/test/htmlelement/querySelectorAll.test.ts [hunk 1]
```diff
@@ -39,6 +39,23 @@ test('test Element.querySelectorAll on id selectors', (t) => {
   t.deepEqual(document.querySelectorAll(`#${DIV_ID}`), [div]);
 });
 
+// TODO: create better solution for parameterizing server tests.
+test.serial('server=true: test Element.querySelectorAll on id selectors', (t) => {
+  process.env.SERVER = 'true';
```
Comment (rcebulko) [inline src/test/htmlelement/querySelectorAll.test.ts:44]: not sure what testing framework is in use, but maybe something like `sandbox.stub(process.env, 'SERVER').value('true');` or something similar could do the trick.

File: src/test/htmlelement/querySelectorAll.test.ts [hunk 2]
```diff
@@ -39,6 +39,23 @@ test('test Element.querySelectorAll on id selectors', (t) => {
...
+  try {
+    const document = createTestingDocument();
+    const testEl = document.createElement('test-el');
+    const id = 'TEST_EL_ID';
+    testEl.setAttribute('id', id);
+    document.appendChild(testEl);
+
+    const result = document.querySelectorAll(`#${id}`);
```
Comment (rcebulko) [inline src/test/htmlelement/querySelectorAll.test.ts:52]: nit: I'm usually a fan of DRY "do repeat yourself" for tests, in this case replicating the string in multiple places

**Review by rcebulko — APPROVED** [review]

File: src/test/htmlelement/querySelectorAll.test.ts (hunk 2 above)
Comment (samouri) [reply src/test/htmlelement/querySelectorAll.test.ts:52]: I'd call that WET! I'll update

File: src/test/htmlelement/querySelectorAll.test.ts (hunk 1 above)
Comment (samouri) [reply src/test/htmlelement/querySelectorAll.test.ts:44]: Will investigate this for a future PR

**Pull request merged**